src/NuGet.Core/NuGet.Build.Tasks.Pack/PackTaskLogic.cs

```diff
@@ -432,16 +440,25 @@ private IEnumerable<OutputLibFile> InitLibFiles(IMSBuildItem[] libFiles)
                     targetPath = Path.GetFileName(finalOutputPath);
                 }
 
-                if (string.IsNullOrEmpty(targetFramework) || NuGetFramework.Parse(targetFramework).IsSpecificFramework == false)
+                NuGetFramework fw = NuGetFramework.Parse(tfm);
+                var isNet5EraTfm = fw.Version.Major >= 5 && StringComparer.OrdinalIgnoreCase.Equals(FrameworkConstants.FrameworkIdentifiers.NetCoreApp, fw.Framework);
+                if (isNet5EraTfm)
                 {
-                    throw new PackagingException(NuGetLogCode.NU5027, string.Format(CultureInfo.CurrentCulture, Strings.InvalidTargetFramework, finalOutputPath));
+                    var dotIdx = tfm.IndexOf('.');
+                    var dashIdx = tfm.IndexOf('-');
+                    var isDottedFwVersion = (dashIdx > -1 && dotIdx > -1 && dotIdx < dashIdx) || (dashIdx == -1 && dotIdx > -1);
+                    if (!isDottedFwVersion)
+                    {
+
+                        logger.LogWarning(string.Format(CultureInfo.CurrentCulture, Strings.MissingRequiredDot, tfm));
```

Comment:
We *have* to validate it sooner, because for some scenarios, we actually parse and serialize the TFM before it ever gets to the Rules. So it had to go in both places :\

Reply:
Why are we validating it in the first place?
Shouldn't we only validate it in the folder name?